shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/network/core/TokenizedNetworkClientTest.kt

```diff
@@ -0,0 +1,90 @@
+package co.nimblehq.blisskmmic.data.network.core
+
+import co.nimblehq.blisskmmic.BuildKonfig
+import co.nimblehq.blisskmmic.data.database.datasource.LocalDataSource
+import co.nimblehq.blisskmmic.data.database.datasource.MockLocalDataSource
+import co.nimblehq.blisskmmic.data.database.model.TokenDatabaseModel
+import co.nimblehq.blisskmmic.data.network.helpers.API_VERSION
+import co.nimblehq.blisskmmic.helpers.mock.NETWORK_MOCK_MODEL_RESULT
+import co.nimblehq.blisskmmic.helpers.mock.NetworkMockModel
+import co.nimblehq.blisskmmic.helpers.mock.ktor.jsonTokenizedMockEngine
+import co.nimblehq.jsonapi.model.JsonApiException
+import io.kotest.matchers.collections.shouldContain
+import io.kotest.matchers.shouldBe
+import io.ktor.client.request.*
+import io.ktor.http.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.test.runTest
+import org.kodein.mock.Mocker
+import org.kodein.mock.UsesMocks
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.fail
+
+@ExperimentalCoroutinesApi
+@UsesMocks(LocalDataSource::class)
+class TokenizedNetworkClientTest {
+
+    private val mocker = Mocker()
+    private val localDataSource = MockLocalDataSource(mocker)
+
+    private val token = TokenDatabaseModel(
+        "Access",
+        "",
+        1,
+        "",
+        1
+    )
+    private val path = "user"
+    private val request = HttpRequestBuilder()
+
+    @BeforeTest
+    fun setUp() {
+        mocker.reset()
+        request.url("$BuildKonfig.BASE_URL$API_VERSION$path")
+    }
+
+    @Test
+    fun `when calling fetch, it returns correct object`() = runTest {
+        mocker.every {
+            localDataSource.getToken()
+        } returns flow { emit(token) }
```

Comment:
We can use `flowOf` instead of `flow { emit(...) }`
```suggestion
        } returns flowOf(token)
```